templates/scripts/monitor.sh.j2

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+# vim: set ft=sh:
+# Script that updates Consul service metadata.
+
+CONSUL_SERVICE_FILE='/etc/consul/service_{{ beacon_node_consul_service_file_name }}.json'
+
+# Not pretty and will break. Wish we had --version-json.
+function get_spec()     { echo "${BN_VERSION}" | grep 'specification' | cut -d' ' -f3; }
+function get_version()  { echo "${BN_VERSION}" | grep 'beacon node'   | cut -d' ' -f4; }
+function get_compiler() { echo "${BN_VERSION}" | grep 'Compiler'      | cut -d' ' -f4; }
+
+function date_iso8691() { date -u +"%Y-%m-%dT%H:%M:%SZ"; }
+
+function update_service_json() {
+    # Uglier but MUCH less effort than using the API.
+    sed "${CONSUL_SERVICE_FILE}" \
+        -e 's#"spec": ".*"#"spec": "'$(get_spec)'"#' \
+        -e 's#"version": ".*"#"version": "'$(get_version)'"#' \
+        -e 's#"compiler": ".*"#"compiler": "'$(get_compiler)'"#' \
+        | sponge "${CONSUL_SERVICE_FILE}"
```

Comment:
Is it better than `sed -i`?

Reply:
Well, it depends. Sed by default creates a temporary file and then renames it.
```
 > echo 'wat' > test                                               
 > strace -e trace=%file sed -i 's/wat/wtf/' test 2>&1 | grep rename         
rename("./sedM3z7C6", "test")           = 0
```
Using `sponge` skips that, which is less I/O and also less required permissions.

Reply:
But it's not really that of a big difference. I used it because I was trying to use most strict permissions using `ProtectSystem` setting in service definition before, but settled on the current setup.